src/components/Top-Bar.vue

```diff
@@ -59,8 +70,14 @@ export default {
     }
   },
   computed: {
+    ...mapGetters(['apps']),
+
     isPublicPage () {
       return !this.userId
+    },
+
+    currentExtensionName () {
+      return this.$gettext(this.apps[this.currentExtension].name)
```

Comment:
For apps provided in the `applications` part of config.json (see sample config), the extension doesn't live in `this.apps`. I've looked into the available data in store and hacked this (dirty):

```
if (this.apps[this.currentExtension]) {
        return this.$gettext(this.apps[this.currentExtension].name)
      } else {
        const applications = this.configuration.applications.filter(app => {
          if (app.path) {
            return app.path.includes(this.currentExtension)
          } else {
            return app.url.includes(this.currentExtension)
          }
        })
        if (applications) {
          // TODO: get title for correct lang or fallback lang from first application's title
          return 'App Title'
        } else {
          // TODO: I don't know
          return 'App Title'
        }
      }
``` 

You have to add `'configuration'` to `mapGetters` array as well.

Reply:
I hope that there is an easier way to select the correct application from `configuration.applications`, because to be honest, my filter also looks hacky as hell.

Reply:
As discussed, applications list in config is meant to include extensions which live in their own SPAs. This will split the responsibility for displaying the current extension. Let's see after we have such an extension as an example how we need to implement this.

Reply:
I'll raise an issue for it so we don't forget it later